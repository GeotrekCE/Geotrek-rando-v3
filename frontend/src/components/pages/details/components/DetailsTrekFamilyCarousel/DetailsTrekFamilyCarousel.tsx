import Slider, { CustomArrowProps } from 'react-slick';
import styled, { css } from 'styled-components';
import { colorPalette, desktopOnly, getSpacing } from 'stylesheet';
import { TrekChild } from 'modules/details/interface';
import { Link } from 'components/Link';
import { generateChildrenDetailsUrl } from '../../utils';

interface DetailsTrekFamilyCarouselProps {
  trekId: string;
  parentId: string;
  trekChildren: TrekChild[];
}

export const DetailsTrekFamilyCarousel: React.FC<DetailsTrekFamilyCarouselProps> = ({
  trekChildren,
  trekId,
  parentId,
}) => {
  const currentTrekChild = trekChildren.find(trek => trek.id === trekId);
  return (
    <Slider
      speed={500}
      infinite={false}
      nextArrow={<NextArrow />}
      prevArrow={<PrevArrow />}
      swipe={false}
      slidesToShow={3}
      slidesToScroll={2}
      initialSlide={currentTrekChild ? currentTrekChild.rank : 0}
      className="mb-3 desktop:mb-6 mx-5 desktop:mr-10 desktop:ml-8"
    >
      {trekChildren.map((trekChild, i) => (
        <div key={i} className="outline-none">
          <Link href={generateChildrenDetailsUrl(trekChild.id, trekChild.name, parentId)}>
            <div
              className={`truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all ${
                trekChild.id === trekId ? currentChildClassName : otherChildClassName
              }`}
            >
              <span>{`${trekChild.rank}. ${trekChild.name}`}</span>
            </div>
          </Link>
        </div>
      ))}
    </Slider>
  );
};

const otherChildClassName = 'border-greySoft text-primary3 bg-white';
const currentChildClassName = 'border-primary1 text-white bg-primary1';

const PrevArrow = (props: CustomArrowProps) => {
  const { className, onClick } = props;
  return <StyledLeftArrow className={className} onClick={onClick} />;
};

const NextArrow = (props: CustomArrowProps) => {
  const { className, onClick } = props;
  return <StyledRightArrow className={className} onClick={onClick} />;
};

const StyledArrow = styled.div`
  z-index: 10;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  transition-property: opacity;
  transition-duration: 300ms;
  &:hover {
    opacity: 1;
  }
  &::before {
    content: 'âžœ';
    opacity: 0.85;
    display: flex;
    justify-content: center;
    align-items: baseline;
    color: ${colorPalette.primary3};
    background-color: white;
    height: ${getSpacing(6)};
    width: ${getSpacing(6)};
    flex: none;
    border-radius: ${getSpacing(4)};
    border: 1px solid ${colorPalette.greySoft};
    font-size: 18px;
    ${desktopOnly(css`
      font-size: 25px;
      height: ${getSpacing(8)};
      width: ${getSpacing(8)};
    `)}
  }
`;
const StyledRightArrow = styled(StyledArrow)`
  right: -${getSpacing(6)};
  ${desktopOnly(css`
    right: -${getSpacing(8)};
  `)}
`;

const StyledLeftArrow = styled(StyledArrow)`
  left: -${getSpacing(6)};
  &::before {
    left: -${getSpacing(8)};
    transform: rotate(180deg);
  }
`;
