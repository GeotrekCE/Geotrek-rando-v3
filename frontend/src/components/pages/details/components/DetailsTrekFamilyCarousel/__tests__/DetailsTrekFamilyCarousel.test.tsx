import { render } from 'services/testing/reactTestingLibraryWrapper';
import { DetailsTrekFamilyCarousel } from '../DetailsTrekFamilyCarousel';
describe('DetailsTrekFamilyCarousel', () => {
  it('should display a DetailsTrekFamilyCarousel', () => {
    const propsTrekFamilyCarousel = {
      trekId: '604',
      parentId: '501',
      trekChildren: [
        {
          id: '596',
          rank: 1,
          name: 'Boucle de malade',
        },
        {
          id: '604',
          rank: 2,
          name: 'Deuxième étape',
        },
        {
          id: '2',
          rank: 3,
          name: 'Troisième étape',
        },
        {
          id: '4',
          rank: 4,
          name: 'Quatrième étape',
        },
        {
          id: '5',
          rank: 5,
          name: 'Encore une étape',
        },
        {
          id: '6',
          rank: 6,
          name: "C'est la fin !",
        },
      ],
    };
    const component = render(<DetailsTrekFamilyCarousel {...propsTrekFamilyCarousel} />);

    expect(component).toMatchSnapshot();
  });
});
