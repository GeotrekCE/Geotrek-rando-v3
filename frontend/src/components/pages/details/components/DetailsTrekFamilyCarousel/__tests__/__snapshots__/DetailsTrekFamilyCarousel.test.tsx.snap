// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsTrekFamilyCarousel should display a DetailsTrekFamilyCarousel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="slick-slider my-1.5 desktop:my-3 mx-7 desktop:mr-10 desktop:ml-8 slick-initialized"
        dir="ltr"
      >
        <div
          class="sc-dlfnuX sc-eCstlR iEzKbz fPgiUf slick-arrow slick-prev slick-disabled"
        />
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="outline-none align-middle py-1"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <a
                    class="sc-bdfBQB btWAAI"
                    href="/trek/596-Boucle-de-malade?parentId=501"
                  >
                    <div
                      class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                    >
                      <span>
                        1. Boucle de malade
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="outline-none align-middle py-1"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <a
                    class="sc-bdfBQB btWAAI"
                    href="/trek/604-Deuxieme-etape?parentId=501"
                  >
                    <div
                      class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-primary1 text-white bg-primary1"
                    >
                      <span>
                        2. Deuxième étape
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="2"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="outline-none align-middle py-1"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <a
                    class="sc-bdfBQB btWAAI"
                    href="/trek/2-Troisieme-etape?parentId=501"
                  >
                    <div
                      class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                    >
                      <span>
                        3. Troisième étape
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="3"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="outline-none align-middle py-1"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <a
                    class="sc-bdfBQB btWAAI"
                    href="/trek/4-Quatrieme-etape?parentId=501"
                  >
                    <div
                      class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                    >
                      <span>
                        4. Quatrième étape
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="4"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="outline-none align-middle py-1"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <a
                    class="sc-bdfBQB btWAAI"
                    href="/trek/5-Encore-une-etape?parentId=501"
                  >
                    <div
                      class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                    >
                      <span>
                        5. Encore une étape
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="5"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="outline-none align-middle py-1"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <a
                    class="sc-bdfBQB btWAAI"
                    href="/trek/6-C-est-la-fin-!?parentId=501"
                  >
                    <div
                      class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                    >
                      <span>
                        6. C'est la fin !
                      </span>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-dlfnuX sc-hKgJUU iEzKbz fHRbop slick-arrow slick-next"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="slick-slider my-1.5 desktop:my-3 mx-7 desktop:mr-10 desktop:ml-8 slick-initialized"
      dir="ltr"
    >
      <div
        class="sc-dlfnuX sc-eCstlR iEzKbz fPgiUf slick-arrow slick-prev slick-disabled"
      />
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="outline-none align-middle py-1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="sc-bdfBQB btWAAI"
                  href="/trek/596-Boucle-de-malade?parentId=501"
                >
                  <div
                    class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                  >
                    <span>
                      1. Boucle de malade
                    </span>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="outline-none align-middle py-1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="sc-bdfBQB btWAAI"
                  href="/trek/604-Deuxieme-etape?parentId=501"
                >
                  <div
                    class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-primary1 text-white bg-primary1"
                  >
                    <span>
                      2. Deuxième étape
                    </span>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="outline-none align-middle py-1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="sc-bdfBQB btWAAI"
                  href="/trek/2-Troisieme-etape?parentId=501"
                >
                  <div
                    class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                  >
                    <span>
                      3. Troisième étape
                    </span>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="outline-none align-middle py-1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="sc-bdfBQB btWAAI"
                  href="/trek/4-Quatrieme-etape?parentId=501"
                >
                  <div
                    class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                  >
                    <span>
                      4. Quatrième étape
                    </span>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="4"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="outline-none align-middle py-1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="sc-bdfBQB btWAAI"
                  href="/trek/5-Encore-une-etape?parentId=501"
                >
                  <div
                    class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                  >
                    <span>
                      5. Encore une étape
                    </span>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="5"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="outline-none align-middle py-1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <a
                  class="sc-bdfBQB btWAAI"
                  href="/trek/6-C-est-la-fin-!?parentId=501"
                >
                  <div
                    class="truncate px-2 desktop:px-4 py-2
              mx-2p desktop:mx-1.5
              text-P3 desktop:text-P1
              outline-none border border-solid rounded-full
              cursor-pointer hover:shadow-sm transition-all border-greySoft text-primary3 bg-white"
                  >
                    <span>
                      6. C'est la fin !
                    </span>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-dlfnuX sc-hKgJUU iEzKbz fHRbop slick-arrow slick-next"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
